test/unit/logic/delegate.js

```diff
@@ -0,0 +1,535 @@
+				expect(err).to.not.exist;
+				expect(returnedTrs).to.equal(trs);
+				accountsMock.getAccount.verify();
+				done();
+			});
+		});
+	});
+
+	describe.skip('process', function () {
+
+		it('should call the callback', function (done) {
+			delegate.process(trs, sender, done);
+		});
+	});
+
+	describe.skip('getBytes', function () {
+
+		it('should return null when username is empty', function () {
+			delete trs.asset.delegate.username;
+
+			expect(delegate.getBytes(trs)).to.eql(null);
+		});
+
+		it('should return bytes for signature asset', function () {
+			var delegateBytes = delegate.getBytes(trs);
+			expect(delegateBytes.toString()).to.equal(trs.asset.delegate.username);
+		});
+	});
+
+	describe.skip('apply', function () {
+
+		var dummyBlock = {
+			id: '9314232245035524467',
+			height: 1
+		};
+
+		it('should call accounts.setAccountAndGet module with correct parameters', function (done) {
+			accountsMock.setAccountAndGet.once().withExactArgs({
+				address: sender.address,
+				u_isDelegate: 0,
+				isDelegate: 1,
+				vote: 0,
+				u_username: null,
+				username: trs.asset.delegate.username
+			}, dummyCb);
+
+			delegate.apply(trs, dummyBlock, sender, dummyCb);
+			accountsMock.setAccountAndGet.verify();
+
+			done();
+		});
+	});
+
+	describe.skip('undo', function () {
+
+		var dummyBlock = {
+			id: '9314232245035524467',
+			height: 1
+		};
+
+		it('should call accounts.setAccountAndGet module with correct parameters', function (done) {
+			accountsMock.setAccountAndGet.once().withExactArgs({
+				address: sender.address,
+				u_isDelegate: 1,
+				isDelegate: 0,
+				vote: 0
+			}, dummyCb);
+
+			delegate.undo(trs, dummyBlock, sender, dummyCb);
+			accountsMock.setAccountAndGet.verify();
+
+			done();
+		});
+
+		it('should update username to null if sender was not assigned a username before', function (done) {
+			delete sender.username;
+			sender.nameexist = 0;
+
+			accountsMock.setAccountAndGet.once().withExactArgs({
+				address: sender.address,
+				u_isDelegate: 1,
+				isDelegate: 0,
+				vote: 0,
+				username: null,
+				u_username: trs.asset.delegate.username
+			}, dummyCb);
+
+			delegate.undo(trs, dummyBlock, sender, dummyCb);
+			accountsMock.setAccountAndGet.verify();
+
+			done();
+		});
+	});
+
+	describe.skip('applyUnconfirmed', function () {
+
+		it('should call accounts.setAccountAndGet module with correct parameters', function (done) {
+			delete sender.username;
+			sender.nameexist = 0;
+
+			accountsMock.setAccountAndGet.once().withExactArgs({
+				address: sender.address,
+				u_isDelegate: 1,
+				isDelegate: 0,
+				username: null,
+				u_username: trs.asset.delegate.username
+			}, dummyCb);
+
+			delegate.applyUnconfirmed(trs, sender, dummyCb);
+			accountsMock.setAccountAndGet.verify();
+
+			done();
+		});
+	});
+
+	describe.skip('undoUnconfirmed', function () {
+
+		it('should update username to null if account did not have a username before', function (done) {
```

Comment:
isn't it the other way around? It should nullify the username if an account had it before. Also please add separate tests for checking if both - u_username and username fields were nullified.
Please consider the pattern introduced in skeletons PR's - 
```
describe('when trs.asset.delegate.username exists' ...

describe('when trs.asset.delegate.username does not exist' ...
```

Reply:
applyUnconfirmed, undoUnconfirmed, apply, undo are all called after `objectNormalize` and `verify` functions which check for the validity of delegate transaction asset. My test descriptions were vague (which I am updating), but having `describe` block for when trs.asset.delegate.username does not exist seem to me like an overkill.